### PR TITLE
Add reference docs for preview features

### PR DESCRIPTION
**Pull request by BillWagner**
Contributes to: #34529, #34655, #34530

Adding initial content for all three features to make sure some content is live for the upcoming preview. Later PRs will add tutorials and compiler error / warning messages, as needed.

A number of content updates here, covering three features:  *Primary constructors*, *Alias any type*, and *Default lambda parameters*.

- [x] Publish speclets on docs
- [x] Add what's new page for C# 12
- [x] Language reference updates for Primary constructors:
  - Add section on *Instance constructors* describing primary constructors.
  - Link to new section from `struct`, `class`, and `record` pages
- [x] Language reference for alias any type
  - Additional text on `using` aliases. This is small, because the docs didn't explicitly mention what wasn't allowed.
- [x] Language reference for default parameters on lambda expressions
  - Add text on the lambda expressions page to describe the declarations, and their type.
   

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/language-reference/builtin-types/record.md](https://github.com/dotnet/docs/blob/64c17455926739f69b9adb21afc0d0bb8cdc083a/docs/csharp/language-reference/builtin-types/record.md) | [Records (C# reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record?branch=pr-en-us-34876) |
| [docs/csharp/language-reference/builtin-types/struct.md](https://github.com/dotnet/docs/blob/64c17455926739f69b9adb21afc0d0bb8cdc083a/docs/csharp/language-reference/builtin-types/struct.md) | [Structure types (C# reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/struct?branch=pr-en-us-34876) |
| [docs/csharp/language-reference/keywords/using-directive.md](https://github.com/dotnet/docs/blob/64c17455926739f69b9adb21afc0d0bb8cdc083a/docs/csharp/language-reference/keywords/using-directive.md) | [using directive](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-directive?branch=pr-en-us-34876) |
| [docs/csharp/language-reference/operators/lambda-expressions.md](https://github.com/dotnet/docs/blob/64c17455926739f69b9adb21afc0d0bb8cdc083a/docs/csharp/language-reference/operators/lambda-expressions.md) | [Lambda expressions and anonymous functions](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/lambda-expressions?branch=pr-en-us-34876) |
| [docs/csharp/programming-guide/classes-and-structs/constructors.md](https://github.com/dotnet/docs/blob/64c17455926739f69b9adb21afc0d0bb8cdc083a/docs/csharp/programming-guide/classes-and-structs/constructors.md) | [Constructors (C# programming guide)](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/constructors?branch=pr-en-us-34876) |
| [docs/csharp/programming-guide/classes-and-structs/instance-constructors.md](https://github.com/dotnet/docs/blob/64c17455926739f69b9adb21afc0d0bb8cdc083a/docs/csharp/programming-guide/classes-and-structs/instance-constructors.md) | [Instance constructors (C# programming guide)](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/instance-constructors?branch=pr-en-us-34876) |
| [docs/csharp/whats-new/csharp-11.md](https://github.com/dotnet/docs/blob/64c17455926739f69b9adb21afc0d0bb8cdc083a/docs/csharp/whats-new/csharp-11.md) | [What's new in C# 11](https://review.learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-11?branch=pr-en-us-34876) |
| [docs/csharp/whats-new/csharp-12.md](https://github.com/dotnet/docs/blob/64c17455926739f69b9adb21afc0d0bb8cdc083a/docs/csharp/whats-new/csharp-12.md) | [docs/csharp/whats-new/csharp-12](https://review.learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-12?branch=pr-en-us-34876) |

### CODE REVIEW

**Comment by BillWagner** [comment]
Current warnings addressed in dotnet/csharplang#7093

File: docs/csharp/programming-guide/classes-and-structs/instance-constructors.md [hunk 1]
```diff
@@ -34,6 +33,22 @@ A *structure* type always provides a parameterless constructor as follows:
...
+
+:::code language="csharp" source="./snippets/instance-constructors/widgets/Program.cs" id="DerivedPrimaryConstructor":::
+
+For non-`record` types, if a primary constructor is read or written in the body of the type, the compiler captures the parameter in a private field. This private field has a compiler generate name that is different from the primary constructor parameter. That ensures the private field name doesn't conflict with other variable names. It's also harder to find using Reflection. If a primary constructor parameter isn't used in the body of the type, no private field is captured. This prevents accidentally allocating two copies of a primary constructor parameter that is passed to a base constructor.
```
Comment (pkulikov) [inline docs/csharp/programming-guide/classes-and-structs/instance-constructors.md:48]:
```suggestion
For non-`record` types, if a primary constructor is read or written in the body of the type, the compiler captures the parameter in a private field. This private field has a compiler-generated name that is different from the primary constructor parameter. That ensures the private field name doesn't conflict with other variable names. It's also harder to find using reflection. If a primary constructor parameter isn't used in the body of the type, no private field is captured. This prevents accidentally allocating two copies of a primary constructor parameter that is passed to a base constructor.
```

File: docs/csharp/programming-guide/classes-and-structs/instance-constructors.md [hunk 2]
```diff
@@ -34,6 +33,22 @@ A *structure* type always provides a parameterless constructor as follows:
...
+
+For `record` types (`record class`, and `record struct`), the compiler generates public properties for each of the positional parameters in the primary constructor. For `record class` types, if a primary constructor parameter uses the same name as a base primary constructor, that property is a public property of the base `record class` type. It is not duplicated in the derived `record class` type. These properties aren't generated for non-`record` types.
```
Comment (pkulikov) [inline docs/csharp/programming-guide/classes-and-structs/instance-constructors.md:50]:
```suggestion
For `record` types (`record class` and `record struct`), the compiler generates public properties for each of the positional parameters in the primary constructor. For `record class` types, if a primary constructor parameter uses the same name as a base primary constructor, that property is a public property of the base `record class` type. It is not duplicated in the derived `record class` type. These properties aren't generated for non-`record` types.
```

**Review by IEvangelist — APPROVED** [review]
Looks good to me.

**Pull request merged**